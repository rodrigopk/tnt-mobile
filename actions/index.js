export const DATA_AVAILABLE = 'DATA_AVAILABLE';

//Import the sample data
import Data from '../instructions.json';

export function getData(){
    return (dispatch) => {

        setTimeout(() => {
            var message  = Data.message;
            dispatch({type: DATA_AVAILABLE, message: message});
        }, 2000);

    };
}
