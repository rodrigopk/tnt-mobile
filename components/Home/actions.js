export const DATA_AVAILABLE = 'MESSAGE_RECEIVED';

//Import the sample data
import Data from '../../hello.json';

export const fetchHelloMessage = () => {
    return (dispatch) => {

        setTimeout(() => {
            var message  = Data.message;
            dispatch({type: DATA_AVAILABLE, message: message});
        }, 2000);

    };
};

export default { fetchHelloMessage };
