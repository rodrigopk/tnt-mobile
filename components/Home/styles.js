import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  activityIndicatorContainer:{
    backgroundColor: "#fff",
    alignItems: 'center',
    justifyContent: 'center',
    flex: 1,
  },
  messageContainer: {
    flex:1,
    backgroundColor: '#F5F5F5',
    paddingTop:20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  message:{
    fontSize: 21,
  },
});

export default styles;
