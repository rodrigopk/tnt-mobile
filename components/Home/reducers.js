import { combineReducers } from 'redux';

import { MESSAGE_RECEIVED } from "./actions"

let initialState = {
  message: '',
  title: '',
  loading:true,
};

const homeReducer = (state = initialState, action) => {
    switch (action.type) {
        case MESSAGE_RECEIVED:
            state = Object.assign(
              {},
              state,
              {
                title: action.title,
                message: action.message,
                loading:false,
              });
            return state;
        default:
            return state;
    }
};

export default homeReducer;
