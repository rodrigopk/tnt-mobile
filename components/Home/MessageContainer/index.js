import React, { Component } from 'react';
import { View, Text } from 'react-native';

import styles from './styles';

const MessageContainer = ({ title, message }) => (
  <View style={styles.container}>
    <Text style={styles.title}>
      { title }
    </Text>
    <Text style={styles.message}>
      { message }
    </Text>
  </View>
);

export default MessageContainer;
