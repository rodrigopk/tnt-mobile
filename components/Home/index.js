import React, { Component } from 'react';
import {
  View,
  Text,
  ActivityIndicator,
} from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { fetchHelloMessage } from './actions';
import styles from './styles';

class Home extends Component {
  componentDidMount() {
    this.props.getMessage();
  }

  render() {
    if (this.props.loading) {
      return (
          <View style={styles.activityIndicatorContainer}>
              <ActivityIndicator
                  animating={true}
                  style={[{height: 80}]}
                  size="small"
              />
          </View>
      );
    } else {
      return (
        <View style={styles.messageContainer}>
          <Text style={styles.message}>
            { this.props.message }
          </Text>
        </View>
      );
    }
  }
};

const mapStateToProps = state => ({
  loading: state.home.loading,
  message: state.home.message,
});

const mapDispatchToProps = dispatch => ({
  getMessage: () =>  dispatch(fetchHelloMessage()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
