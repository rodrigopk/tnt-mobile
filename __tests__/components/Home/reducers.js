import reducer from '../../../components/Home/reducers';
import { MESSAGE_RECEIVED } from "../../../components/Home/actions"



describe('Home reducer', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('MESSAGE_RECEIVED action', () => {
    const action = {
      type: MESSAGE_RECEIVED,
      title: 'Some title',
      message: 'Some message',
    };

    expect(reducer(undefined, action)).toMatchSnapshot();
  });
});
