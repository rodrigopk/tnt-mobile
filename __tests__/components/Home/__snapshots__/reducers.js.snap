// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home reducer MESSAGE_RECEIVED action 1`] = `
Object {
  "loading": false,
  "message": "Some message",
  "title": "Some title",
}
`;

exports[`Home reducer should return initial state 1`] = `
Object {
  "loading": true,
  "message": "",
  "title": "",
}
`;
